### DISCUSSION TITLE
Frage zur Verdrahtung RX-Ports

### DISCUSSION THREAD

**Discussion by hoschiking** [softwarecrash/Victron2MQTT#26] [General]
gemäß deines Wiring-Diagrams soll ja zum Steuern des MPPT der Pin D6 mit dem RX Port verbunden werden.
Jetzt ist es bekannt, dass man die Schnittstelle in den meisten Fällen galvanisch trennen sollte, sonst kann es böse enden.
Falls man die galvanische Trennung einbaut, wie hast du den Port verdrahtet, damit die galvanische Tennung erhalten bleibt?

**Comment by softwarecrash** [comment]
Schau dir den ADUM1201 an

**Comment by all-solutions** [comment]
Und so ganz nebenbei. Wenn Du den ESP über ein USB-Netzteil (mit **einem** USB-Port, nur für den Wemos) versorgst, hast Du die galvanische Trennung im Netzteil. Da kann der ESP ruhig "ungetrennt" sein, da er dann sonst keine elektrische Verbindung zu anderen Sachen hat.

**Comment by hoschiking** [comment]
Korrekt, jedoch wird es wohl bei den wenigsten der Fall sein. Bei mir kommt noch ein Raspi, ein ESP32 mit OpenDTU,  ein ESP 8266 mit Tasmota usw dazu. Ich betreibe alles direkt am 24V Speicher, dann spare ich mir die ganzen Steckernetzteile und habe direkt für den Raspi ein usv kostenlos dabei. Da muss man bei den vielen Schnittstellen peinlichst auf die galvanische Trennung achten, sonst brät man sich andauernd die Elektronik kaputt. Der Speicher hat ein JK BMS, das blöderweise, wie viele andere auch, beim Abschalten den Minus trennt, und genau das ist das Problem. N Kanal MOSFET haben nun Mal wesentlich kleineren R DS on.
Ich werde noch einen 4ten MPPT anschließen, da werde ich dieses Projekt zum ersten Mal testen. Daher schade, dass nur ein MPPT unterstützt wird. Ich bräuchte 4 ESP dafür....
Die bisherigen 3 laufen über OpenDTU ON battery, das kann 3 MPPT gleichzeitig ins mqtt integrieren

**Reply by softwarecrash** [reply]
Dieses Projekt kann eben nur ein gerät, das wird sich auch vorraussichtlich nicht ändern.
Dann nimm doch venusos oder cerbos wenn dir 3 esp01 zu teuer sind

**Reply by hoschiking** [reply]
Das hat doch mit zu teuer nichts zu tun.

**Reply by all-solutions** [reply]
Du bist der absolute Kandidat für einen GND-Loop.
Und was machst Du, wenn das BMS warum auch immer abschaltet? Alle Geräte "dunkel". Und jetzt sag mir nicht, dass Du die 24V VOR dem BMS abgreifst.
Das Projekt ist mitunter auf ein Gerät ausgelegt, damit man sich keine GND-Loops einbauen kann. Es gibt nunmal auch Leute, deren das Fachwissen fehlt, trotzdem aber eine Anlage haben und dann dafür auch dieses Projekt nutzen. Und ja, ich rede da aus Erfahrung und den dementsprechenden Feedbacks bei Anfragen und Support meinerseits.
Und ganz ehrlich: mach doch bei OpenDTU ein Feature-Request, dass die mehr als 3 Geräte unterstützen. Denke, auch sie haben ihre Gründe, es auf 3 zu begrenzen.

**Reply by hoschiking** [reply]
Ich hatte gerade dargelegt, wie ich es gelöst habe, damit genau das nicht passiert.
Jede Datenleitung ist galvanisch getrennt, daher sehe ich da kein Problem.
Warum dieses aggressive Auftreten?
Ich habe doch nichts kritisiert, daher verstehe ich deinen Post nicht. Der ist voller falscher Unterstellungen, daher lassen wir es am besten.

**Reply by softwarecrash** [reply]
Bevor das hier eskaliert schließe ich es.
Zusammenfassend bleibt
- victron2mqtt kann nur ein gerät, wurde an anderer Stelle schon einmal geklärt
- galvanische Trennung -> ADUM1201
